lib/instrumentation/context.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+var { parseUrl } = require('../parsers')
+
+const LEFT_SQUARE_BRACKET = 91 // [
+const RIGHT_SQUARE_BRACKET = 93 // ]
+
+// Get the port number including the default port for a protocols
+function getPortNumber (port, protocol) {
+  if (port === '') {
+    port = protocol === 'http:' ? '80' : protocol === 'https:' ? '443' : ''
+  }
+  return port
+}
+
+function getDestination (url, spanType) {
```

Comment:
Perhaps rename as `getURLDestination`, or `getHTTPDestination`? This is really only appropriate for http(s) spans.